src/types.rs

```diff
@@ -167,6 +169,12 @@ impl<'de> serde::de::Deserialize<'de> for BinaryData<'de> {
     }
 }
 
+#[derive(Debug)]
+pub struct Auth {
+    pub user_name: String<64>,
+    pub password: String<64>,
```

Comment:
Do we have to have these as stored strings? Can we instead just borrow the memory from the user? The problem with this is that it's imposing a 128 byte overhead on all users of Minimq regardless of whether or not they're using auth. Not sure if this matters, but it would be nice to keep our memory overhead to a minimum.

Reply:
I agree. Because I am new in rust I refer to the code you wrote elsewhere:
[src/session_state.rs](https://github.com/quartiq/minimq/blob/8e7e2b8c6faf72b87a630b0c2a0571a8d117f187/src/session_state.rs#L11)

Reply:
You can just take a `&'a str` instead, although this may require you to merge in `main` branch. Then, when constructing the CONNECT packet, you just wrap the &str in the `Utf8String`. If you want to leave this as-is, I can patch it up after the merge as well and tag you for review :)

Reply:
I prefer you to patch it up after the merge. I think you can handle it better than me.

Reply:
Actually, lets leave these as static strings for now. I wonder if the `'buf` lifetime will cause problems for users